### ISSUE TITLE
Ignore a rule for a specific file in the general configuration

### ISSUE THREAD

**Issue by electriquo** [DavidAnson/markdownlint-cli2#272]
Firstly, thank you for this amazing linter 🙏 

Markdownlint supports [inline file comment/configuration](https://github.com/DavidAnson/markdownlint/blob/v0.32.1/README.md#configuration), such as

```markdown
<!-- markdownlint-disable heading-increment -->
```

This disable a specific rule within the file where the comment appears. But, Markdownlint also supports a [configuration to not allow any inline configuration](https://github.com/DavidAnson/markdownlint-cli2/blob/82c791e94c2e2ab2119437d61c722c8e0c20d6c9/test/markdownlint-cli2-yaml-example/.markdownlint-cli2.yaml#L38)

```yaml
noInlineConfig: true
```

Clearly, these 2 configurations are mutually exclusive.
Is there a way to have forbid any inline configuration and ignore a specific rule for a file? For example:

```yaml
noInlineConfig: true
config:
  heading-increment:
    ignores:
      - foo.md
```

**Comment by DavidAnson** [comment]
Something like this has been asked previously: https://github.com/DavidAnson/markdownlint/issues/666

Your situation seems a little bit different since you don't even trust that file not to override its configuration. But my recommendation is the same: multiple passes of CLI2 passing the appropriate configuration each time.

It's a little more effort to set up, but your scenario is good example of why it wouldn't have been enough even if there was support for the feature above.

You should be able to use the --config command-line parameter to customize a specialized run for these "untrusted" files.

**Comment by DavidAnson** [comment]
Building support for nested sub-configuration that is capable of overriding every setting for arbitrary groups of files is complex to design and would take time to implement and test.

Running two passes works today and is just as efficient, or nearly so. It also automatically supports any new settings in the future.

**Comment by DavidAnson** [comment]
No, thank you. I am not interested in making this change for now.

**Comment by SRachamim** [comment]
I wish we had an option to turn off specific rules in specific files. For example - turning of MD003 on `.mdx` files.

Is there a way to do the currently? If not, can we reconsider accepting @electriquo's PR?
